### PR TITLE
events: move EventEmitter to V8 Extras & worker: make MessagePort an EventEmitter always

### PR DESCRIPTION
**Pull request by addaleax**
To make sure `MessagePort` always implements Node’s event emitter API, independent of the used VM context.

The only bit here that hasn’t been approved yet is the addition of error codes in the commit marked `[squash]`. :)

##### Checklist

- [x] `make -j4 test` (UNIX), or `vcbuild test` (Windows) passes
- [x] tests and/or benchmarks are included
- [x] commit message follows [commit guidelines](https://github.com/ayojs/ayo/blob/latest/CONTRIBUTING.md#commit-message-guidelines)

##### Affected core subsystem(s)

worker

### CODE REVIEW

File: lib/extras/events.js [hunk 1]
```diff
@@ -54,8 +54,12 @@ Object.defineProperty(EventEmitter, 'defaultMaxListeners', {
     console;
     // check whether the input is a positive number (whose value is zero or
     // greater and not a NaN).
-    if (typeof arg !== 'number' || arg < 0 || arg !== arg)
-      throw new TypeError('"defaultMaxListeners" must be a positive number');
+    if (typeof arg !== 'number' || arg < 0 || arg !== arg) {
+      const err =
+          new TypeError('The "defaultMaxListeners" argument is out of range');
+      err.code = 'ERR_OUT_OF_RANGE';
```
Comment (TimothyGu) [inline lib/extras/events.js]: I believe you might also want to set `.name = 'TypeError [ERR_OUT_OF_RANGE]'`

**Review by TimothyGu — APPROVED** [review]
LGTM.

**Review by Qard — APPROVED** [review]

File: src/node_extras.cc [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+  Local<Object> binding = context->GetExtrasBindingObject();
+  CHECK(target->Set(context, FIXED_ONE_BYTE_STRING(isolate, "binding"),
+                    binding).FromMaybe(false));
+}
+
+}  // namespace v8_extras
+
+}  // namespace node
+
+NODE_MODULE_CONTEXT_AWARE_BUILTIN(extras, node::v8_extras::Init)
```
Comment (TimothyGu) [inline src/node_extras.cc]: This should be made an internal binding, as V8 extras have some dangerous tools that we don't want exposed.

File: src/node_messaging.cc [hunk 3]
```diff
@@ -657,8 +659,41 @@ MaybeLocal<Function> GetMessagePortConstructor(
...
+    if (!maybe_ctor.ToLocal(&ctor)) return maybe_ctor;
+
+    // Set up EventEmitter inheritance and some default listners.
+    Local<Private> initialized = env->messageport_initialized_private_symbol();
+    if (!ctor->HasPrivate(context, initialized).FromJust()) {
+      Local<Object> extras = context->GetExtrasBindingObject();
+      Local<Value> make_message_port;
+
+      if (!extras->Get(context,
+                      FIXED_ONE_BYTE_STRING(env->isolate(), "makeMessagePort"))
```
Comment (TimothyGu) [inline src/node_messaging.cc]: s/^/ /

**Review by TimothyGu — APPROVED** [review]

File: src/node_extras.cc (hunk 2 above)
Comment (addaleax) [reply src/node_extras.cc]: yup, done!

File: src/node_messaging.cc (hunk 3 above)
Comment (addaleax) [reply src/node_messaging.cc]: done!

**Comment by addaleax** [comment]
Landed in 26590a244513d7debfb1c362105f638108b0f9be, f5f0b624cd5456d586b3053e05266e2bef870389

**Pull request closed without merging**